src/cpp/lpnamer/main/RunProblemGeneration.cpp

```diff
@@ -92,15 +100,139 @@ void RunProblemGeneration(
   Couplings couplings;
   std::string solver_name = "CBC";
   std::vector<ActiveLink> links = linkBuilder.getLinks();
+
+  auto const mps_file_name = xpansion_output_dir / common_lpnamer::MPS_TXT;
+
   LinkProblemsGenerator linkProblemsGenerator(links, solver_name, logger,
                                               log_file_path, zip_mps);
-  linkProblemsGenerator.treatloop(xpansion_output_dir, antares_archive_path,
-                                  couplings);
+  auto files_mapper = FilesMapper(antares_archive_path);
+  auto mpsList = files_mapper.MpsAndVariablesFilesVect();
+
+  auto lpDir_ = xpansion_output_dir / "lp";
+
+  bool use_zip_implementation = true;
+  bool use_file_implementation = false;
+  if (use_zip_implementation) {
+    std::shared_ptr<ArchiveReader> reader =
+        InstantiateZipReader(antares_archive_path);
+    std::shared_ptr<ArchiveWriter> writer;
+    const std::filesystem::path tmpArchivePath = OpenTmpArchive(lpDir_, writer);
+
+    /* Zip problem writer */
+    auto problem_writer =
```

Comment:
writing mps in zip after PGen will be removed see Pr [628](https://github.com/AntaresSimulatorTeam/antares-xpansion/pull/628) and [622](https://github.com/AntaresSimulatorTeam/antares-xpansion/pull/622)